File: src/Sequence/SeqMod.java
package Sequence;

import Sequence.content.SqContent;
import Sequence.core.SqBundle;
import Sequence.core.SqLog;
import arc.util.Log;
import mindustry.mod.Mod;
import mindustry.mod.Mods;

import static mindustry.Vars.mods;

public class SeqMod extends Mod {
    public static final boolean dev = true;
    public static final String repo = "LiuDai2011/Sequence";
    public static Mods.LoadedMod MOD;

    public SeqMod() {
        SqLog.info("Loaded SeqMod constructor.");

        if (dev) {
            Log.level = Log.LogLevel.debug;
            SqLog.warn("In DEBUG mode!");
        }
    }

    public void loadMeta() {
        MOD = mods.getMod(getClass());

        MOD.meta.version = SqBundle.modCat("meta", "version");
        MOD.meta.author = SqBundle.modCat("meta", "author");
        MOD.meta.displayName = SqBundle.modCat("meta", "display-name");
        MOD.meta.description = SqBundle.modCat("meta", "description");
        MOD.setRepo(repo);
    }

    @Override
    public void loadContent() {
        SqLog.info("Loading seq content.");

        loadMeta();
        SqContent.load();
    }
}

File: src/sequence/SeqMod.kt
@file:JvmName("SeqMod")

package sequence

import arc.util.Log
import mindustry.Vars
import mindustry.mod.Mod
import mindustry.mod.Mods.LoadedMod
import sequence.content.SqContent
import sequence.core.SqBundle
import sequence.core.SqLog
import sequence.graphic.SqColor

class SeqMod : Mod() {
    init {
        SqLog.info("Loaded SeqMod constructor.")
        if (dev) {
            Log.level = Log.LogLevel.debug
            SqLog.warn("In DEBUG mode!")
        }
    }

    private fun loadMeta() {
        MOD = Vars.mods.getMod(javaClass)
        MOD.apply {
            meta.version = SqBundle.mod("meta.version")
            meta.author = SqBundle.modFormat("meta.author", SqColor.NedKelly, SqColor.LiuDai)
            meta.displayName = SqBundle.mod("meta.display-name")
            meta.description = SqBundle.mod("meta.description")
            repo = Companion.repo
        }
    }

    override fun loadContent() {
        SqLog.info("Loading seq content.")
        loadMeta()
        SqContent.loadContent()
    }

    companion object {
        const val dev = true
        const val skipChk = false
        const val repo = "LiuDai2011/Sequence"
        lateinit var MOD: LoadedMod
        val MOD_PREFIX: String by lazy { "${MOD.meta.name}-" }
        val MOD_PREFIX_POINT: String by lazy { "${MOD.meta.name}." }

        fun name(s: String) = MOD_PREFIX + s
    }
}
